### PR TITLE
fix(EMS-793): Policy and exports - fix string fields/inputs not saving as empty with save & back functionality

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR changes the "sanitise form data" functions and therefore fixing an issue where a text input would not be sent to the API/saved in the database, if the value has been removed after being submitted with a value. We previously stripped out empty form fields before sending to the API. Also includes some technical improvements.

## Summary of changes
- Update `sanitiseData` function so that it does not return a field that is in a set of specific number fields and has an empty value. Without this, trying to send number fields with an empty string or value of 0, results in API/DB error and the data is not saved. 
  - Created `NUMBER_FIELDS` array of specific number fields that are in the applicatino forms.
  - Created `shouldIncludeAndSanitiseField` check.
  - Improved `shouldChangeToNumber` checks so that all checks are in this function instead of being outside.
- Simplify `hasFormData` function - do not strip out empty form fields and remove unnecessary param checks (typescript handles this).
- Simplfiy `mapSubmittedData` function - remove unnecessary param checks.
- Simplfiy `getValidFields` function - remove unnecessary param checks and add an empty string check.
- Update company details controller tests to use `sanitiseValue, passing the key and value instead of just the value. Also renamed the page variables import/usage.
- Add E2E tests for wiping the "credit period with buyer" field via "save and go back", for both single and multiple contracts.

### CODE REVIEW

File: src/ui/server/helpers/get-valid-fields/index.ts [hunk 1]
```diff
@@ -6,25 +6,19 @@
...
-      Object.keys(formData).forEach((fieldName) => {
-        if (!fieldsWithErrors.includes(fieldName)) {
-          validFields[fieldName] = formData[fieldName];
-        }
-      });
+  Object.keys(formData).forEach((fieldName) => {
+    const fieldValue = formData[fieldName];
 
-      return validFields;
+    if (!fieldsWithErrors.includes(fieldName) && fieldValue !== '') {
```
Comment (Zainzzkk) [inline src/ui/server/helpers/get-valid-fields/index.ts:16]: a comment here please for conditions.  Worth having an `isEmptyString` helper?

File: src/ui/server/helpers/sanitise-data/index.ts [hunk 2]
```diff
@@ -17,9 +22,9 @@ const {
  * @param {String | Number} Field value
  * @returns {Boolean}
  */
-const shouldChangeToNumber = (value: string | number) => {
-  if (typeof value === 'string' && isNumber(value)) {
-    return true;
+const shouldChangeToNumber = (key: string, value: string | number) => {
+  if (key === COMPANY_NUMBER || key === PHONE_NUMBER || value === '') {
```
Comment (Zainzzkk) [inline src/ui/server/helpers/sanitise-data/index.ts:26]: could use `isEmptyString` here

Comment (ttbarnes) [reply src/ui/server/helpers/sanitise-data/index.ts]: Great shout, created and added new helper function

**Review by Zainzzkk — APPROVED** [review]
LGTM

**Pull request merged**